### ISSUE TITLE
QASYMM8 MobileNet 

### ISSUE THREAD

**Issue by WenguoLi** [ARM-software/ComputeLibrary#378]
How to get QASYMM8 MobileNet ？

**Comment by AnthonyARM** [comment]
We've created a [mobilenet QASYMM8 example](https://github.com/ARM-software/ComputeLibrary/blob/master/examples/graph_mobilenet_qasymm8.cpp)

**Comment by WenguoLi** [comment]
Sorry, I did not say it clearly,  I mean How to get QASYMM8 MobileNet  model  files?

**Comment by WenguoLi** [comment]
Thanks!

**Comment by WenguoLi** [comment]
@AnthonyARM ，Thanks！

 In the mobilenet QASYMM8 example, why are there  only the QuantizationInfo  of weight (for example: depth_weights_quant_info and point_weights_quant_info) , without QuantizationInfo of output of ConvolutionLayer  and  QuantizationInfo of Bias ? 
Is there a detailed description of how to implement inference using QASYMM8 model ？

**Comment by qfmy** [comment]
Hi, is there any information or papers about QASYMM8?  @AnthonyARM  @WenguoLi  @GeorgeARM

**Comment by austingg** [comment]
@WenguoLi  mobilenet has only a few of vanila convolution layer, most of conv are implemente with separable conv( depthwise + pointwise) , you may miss the [conv_quantization_info](https://github.com/ARM-software/ComputeLibrary/blob/master/examples/graph_mobilenet_qasymm8.cpp#L52) . As for bias, it use fp32 in the computation, so bias isn't need to be quantizated.

**Comment by WenguoLi** [comment]
Hi, @AnthonyARM @austingg 
I had converted the mobilenet_v1_1.0_224_quant model to numpy arrays by the script [convert_weights.py ](https://github.com/dividiti/ck-request-asplos18-mobilenets-armcl-opencl/blob/master/package/weights-mobilenet-v1-1.0-224/convert_weights.py )  successfully !
But, there are not *_bias.np that used in  [graph_mobilenet_qasymm8.cpp](https://github.com/ARM-software/ComputeLibrary/blob/master/examples/graph_mobilenet_qasymm8.cpp).
How to get *_bias.np these arrays ?

**Comment by austingg** [comment]
@WenguoLi , when the conv is followed by a bn layer, the bias of conv has no use.  Besides, the toco (tf model converter) should have merged the bn into prior conv(params merged into weights and bias), so there should be no bn layer. That is, bn layer can be precomputed.  You may convert the original tensorflow model , not tf quant model. if quant model , it would have conv bias , and no bn layers.

**Comment by AnthonyARM** [comment]
Yes, that's what the example will do by default if it can't find the weights files.

**Comment by WenguoLi** [comment]
@AnthonyARM ,@austingg, Thanks, I See.

**Comment by WenguoLi** [comment]
@AnthonyARM ,@austingg,
I tested the performance of the  graph_mobilenet_qasymm8 with a single thread. Compared with the graph_mobilenet, I found that performance did not improve more. As follows:

```
$ ./graph_mobilenet_qasymm8 0                   
WARNING: linker: ./graph_mobilenet_qasymm8: unsupported flags DT_FLAGS_1=0x8000000
./graph_mobilenet_qasymm8
Usage: ./graph_mobilenet_qasymm8 0 [path_to_data] [npy_input] [labels]
No input provided: using random values
time: 576231 us
Test passed
 
$ ./graph_mobilenet 0 0                         
WARNING: linker: ./graph_mobilenet: unsupported flags DT_FLAGS_1=0x8000000
./graph_mobilenet
Usage: ./graph_mobilenet 0 0 [path_to_data] [image] [labels]
No data folder provided: using random values
time: 609449 us
Test passed
```
Why for this?

**Comment by psyhtest** [comment]
@WenguoLi If you are interested in MobileNets you may wish to take a look at https://github.com/ARM-software/ComputeLibrary/pull/432, adding a new direct convolution kernel optimised for Bifrost. This was used for our [ReQuEST @ ASPLOS'18 submission](https://github.com/dividiti/ck-request-asplos18-mobilenets-armcl-opencl). While so far we focussed on `F32` inference only, the artifact could be extended and optimised for `QASYMM8` inference both on Bifrost and Neon.
